### PR TITLE
Internalclient

### PR DESCRIPTION
**Pull request by MHBauer**
@pmorie @duglin @arschles

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -136,6 +136,13 @@ $(BINDIR)/lister-gen:
 	  --extra-peer-dirs k8s.io/kubernetes/pkg/api,k8s.io/kubernetes/pkg/api/v1,k8s.io/kubernetes/pkg/apis/meta/v1,k8s.io/kubernetes/pkg/conversion,k8s.io/kubernetes/pkg/runtime \
 	  -O zz_generated.conversion
 	# the previous three directories will be changed from kubernetes to apimachinery in the future
+	# Generate the internal clientset (pkg/client/clientset_generated/internalclientset)
+	$(DOCKER_CMD) $(BINDIR)/client-gen \
+		--input-base "github.com/kubernetes-incubator/service-catalog/pkg/apis/" \
+		--input servicecatalog/ \
```
Comment (pmorie) [inline Makefile]: Should we feed the knowledge about the trailing slash into the docs? (maybe you've done this already)

Comment (MHBauer) [inline Makefile]: I will review the existing doc and PR as necessary.

Comment (duglin) [inline Makefile]: what happens w/o the trailing slash?

**Comment by duglin** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (MHBauer) [inline Makefile]:
```
# client-gen --input-base="github.com/kubernetes-incubator/service-catalog/pkg/apis/" --input="servicecatalog" --clientset-path "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/"
F0207 20:05:58.852988    2343 main.go:216] Error: Failed making a parser: unable to add directory "k8s.io/apimachinery/pkg/fields": unable to import "k8s.io/apimachinery/pkg/fields": cannot find package "k8s.io/apimachinery/pkg/fields" in any of:
        /root/go/src/github.com/kubernetes-incubator/service-catalog/vendor/k8s.io/apimachinery/pkg/fields (vendor tree)
        /usr/local/go/src/k8s.io/apimachinery/pkg/fields (from $GOROOT)
        /root/go/src/k8s.io/apimachinery/pkg/fields (from $GOPATH)
goroutine 1 [running]:
k8s.io/kubernetes/vendor/github.com/golang/glog.stacks(0x861b00, 0xc400000000, 0x1e6, 0x23a)
        /root/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:766 +0xa5
k8s.io/kubernetes/vendor/github.com/golang/glog.(*loggingT).output(0x846d20, 0xc400000003, 0xc4216fe480, 0x826815, 0x7, 0xd8, 0x0)
        /root/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:717 +0x337
k8s.io/kubernetes/vendor/github.com/golang/glog.(*loggingT).printf(0x846d20, 0xc400000003, 0x6eb827, 0x9, 0xc427dbdd50, 0x1, 0x1)
        /root/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:655 +0x14c
k8s.io/kubernetes/vendor/github.com/golang/glog.Fatalf(0x6eb827, 0x9, 0xc427dbdd50, 0x1, 0x1)
        /root/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:1145 +0x67
main.main()
        /root/go/src/k8s.io/kubernetes/cmd/libs/go2idl/client-gen/main.go:216 +0x575
```

https://github.com/kubernetes/community/pull/343 addresses.

Comment (duglin) [inline Makefile]: Just wondering... if the trailing slash isn't there you said things will go nuts - would it make sense to submit a PR that fixes the "nuts" aspect? Even if it just adds the trailing / for them when there is no "/" in the string at all.

Comment (MHBauer) [inline Makefile]: https://github.com/kubernetes/community/pull/343#issuecomment-278119968

Comment (MHBauer) [inline Makefile]: https://github.com/kubernetes/kubernetes/pull/41094

**Pull request merged**